### ISSUE TITLE
Bug in 5.0.0 (also 4.3.4): Call of the dialog to create a new column throws null pointer exception

### ISSUE THREAD

**Issue by thiemo-taken** [dbeaver/dbeaver#3111]
![grafik](https://user-images.githubusercontent.com/35094030/37152394-7bc3c6aa-22d9-11e8-95f7-1f228e95815b.png)
![grafik](https://user-images.githubusercontent.com/35094030/37152402-833331f0-22d9-11e8-9857-8d378fd41918.png)

**Comment by serge-rider** [comment]
Can't reproduce (perhaps this error occurs only for some particular table types or something).
Please post error log (you can get it in Error Log view or in log file https://github.com/dbeaver/dbeaver/wiki/Log-files)

**Comment by thiemo-taken** [comment]
### Error log view
#### Message is empty
#### Exception stack
> java.lang.NullPointerException
	at org.jkiss.dbeaver.model.impl.DBObjectNameCaseTransformer.transformName(DBObjectNameCaseTransformer.java:64)
	at org.jkiss.dbeaver.model.impl.sql.edit.struct.SQLTableColumnManager.getNewColumnName(SQLTableColumnManager.java:191)
	at org.jkiss.dbeaver.ext.oracle.edit.OracleTableColumnManager.createDatabaseObject(OracleTableColumnManager.java:66)
	at org.jkiss.dbeaver.ext.oracle.edit.OracleTableColumnManager.createDatabaseObject(OracleTableColumnManager.java:1)
	at org.jkiss.dbeaver.model.impl.sql.edit.SQLObjectEditor.createNewObject(SQLObjectEditor.java:82)
	at org.jkiss.dbeaver.model.impl.sql.edit.SQLObjectEditor.createNewObject(SQLObjectEditor.java:1)
	at org.jkiss.dbeaver.ui.actions.navigator.NavigatorHandlerObjectCreateBase$CreateJob.run(NavigatorHandlerObjectCreateBase.java:142)
	at org.jkiss.dbeaver.model.runtime.AbstractJob.run(AbstractJob.java:95)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:56)

#### Session data
>eclipse.buildId=unknown
java.version=9.0.1
java.vendor=Oracle Corporation
BootLoader constants: OS=win32, ARCH=x86_64, WS=win32, NL=en
Command-line arguments:  -os win32 -ws win32 -arch x86_64

### Log file
>!ENTRY org.jkiss.dbeaver.model 4 0 2018-03-09 08:34:36.038
!MESSAGE 
!STACK 0
java.lang.NullPointerException
	at org.jkiss.dbeaver.model.impl.DBObjectNameCaseTransformer.transformName(DBObjectNameCaseTransformer.java:64)
	at org.jkiss.dbeaver.model.impl.sql.edit.struct.SQLTableColumnManager.getNewColumnName(SQLTableColumnManager.java:191)
	at org.jkiss.dbeaver.ext.oracle.edit.OracleTableColumnManager.createDatabaseObject(OracleTableColumnManager.java:66)
	at org.jkiss.dbeaver.ext.oracle.edit.OracleTableColumnManager.createDatabaseObject(OracleTableColumnManager.java:1)
	at org.jkiss.dbeaver.model.impl.sql.edit.SQLObjectEditor.createNewObject(SQLObjectEditor.java:82)
	at org.jkiss.dbeaver.model.impl.sql.edit.SQLObjectEditor.createNewObject(SQLObjectEditor.java:1)
	at org.jkiss.dbeaver.ui.actions.navigator.NavigatorHandlerObjectCreateBase$CreateJob.run(NavigatorHandlerObjectCreateBase.java:142)
	at org.jkiss.dbeaver.model.runtime.AbstractJob.run(AbstractJob.java:95)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:56)

**Comment by thiemo-taken** [comment]
It is not a particular Oracle table it happens on. E. g.

```SQL
  CREATE TABLE "IL"."CASE" 
   (	"T_RECORD_ID" RAW(16) DEFAULT sys_guid() CONSTRAINT "CASE_T_RECORD_ID_NN" NOT NULL ENABLE, 
	"T_INSERT_ON" TIMESTAMP (6) WITH LOCAL TIME ZONE DEFAULT systimestamp CONSTRAINT "CASE_T_INSERT_ON_NN" NOT NULL ENABLE, 
	"T_LASTMOD_ON" TIMESTAMP (6) WITH LOCAL TIME ZONE DEFAULT systimestamp CONSTRAINT "CASE_T_LASTMOD_ON_NN" NOT NULL ENABLE, 
	"T_SOURCE" VARCHAR2(1000 CHAR) CONSTRAINT "CASE_T_SOURCES_NN" NOT NULL ENABLE, 
	"T_PRIMARY_SYSTEM" VARCHAR2(1000 CHAR), 
	"T_JOB_ID" VARCHAR2(1000 CHAR) CONSTRAINT "CASE_T_JOB_ID_NN" NOT NULL ENABLE, 
	"T_BATCH_ID" VARCHAR2(1000 CHAR) CONSTRAINT "CASE_T_BATCH_IDECNT_NN" NOT NULL ENABLE, 
	"FK_PTNT_T_RECORD_ID" RAW(16) CONSTRAINT "CASE_FK_PTNT_T_RECORD_ID_NN" NOT NULL ENABLE, 
	"FK_MPST_T_RECORD_ID" RAW(16) CONSTRAINT "CASE_FK_MPCT_T_RECORD_ID_NN" NOT NULL ENABLE, 
	"CASE_ID" VARCHAR2(1000 CHAR) NOT NULL ENABLE, 
	"ADMISSION_DATE" TIMESTAMP (6) WITH LOCAL TIME ZONE CONSTRAINT "CASE_ADMISSION_DATE_NN" NOT NULL ENABLE, 
	"DISCHARGE_DATE" TIMESTAMP (6) WITH LOCAL TIME ZONE, 
	"VALID_FROM" TIMESTAMP (6) WITH LOCAL TIME ZONE CONSTRAINT "CASE_VALID_FROM_NN" NOT NULL ENABLE, 
	"VALID_UNTIL" TIMESTAMP (6) WITH LOCAL TIME ZONE CONSTRAINT "CASE_VALID_UNTIL_NN" NOT NULL ENABLE, 
	 CONSTRAINT "CASE_PK" PRIMARY KEY ("T_RECORD_ID")
  USING INDEX PCTFREE 10 INITRANS 2 MAXTRANS 255 COMPUTE STATISTICS NOLOGGING 
  TABLESPACE "CDWH_DAT"  ENABLE, 
	 CONSTRAINT "CASE_02_UN" UNIQUE ("CASE_ID")
  USING INDEX PCTFREE 10 INITRANS 2 MAXTRANS 167 COMPUTE STATISTICS NOLOGGING 
  TABLESPACE "CDWH_DAT"  ENABLE, 
	 CONSTRAINT "CASE_MPCT_FK" FOREIGN KEY ("FK_MPST_T_RECORD_ID")
	  REFERENCES "IL"."MAP_CASE_TYPE" ("T_RECORD_ID") ON DELETE CASCADE ENABLE, 
	 CONSTRAINT "CASE_PTNT_FK" FOREIGN KEY ("FK_PTNT_T_RECORD_ID")
	  REFERENCES "IL"."PATIENT" ("T_RECORD_ID") ON DELETE CASCADE ENABLE
   ) SEGMENT CREATION DEFERRED 
  PCTFREE 10 PCTUSED 40 INITRANS 1 MAXTRANS 255 
 NOCOMPRESS LOGGING
  TABLESPACE "CDWH_DAT" 

   COMMENT ON COLUMN "IL"."CASE"."T_RECORD_ID" IS 'Primary key'
   COMMENT ON COLUMN "IL"."CASE"."T_INSERT_ON" IS 'Point in time when this records was inserted'
   COMMENT ON COLUMN "IL"."CASE"."T_LASTMOD_ON" IS 'Point in time when this records was modified the last time. The insert is considered a modification as well.'
   COMMENT ON COLUMN "IL"."CASE"."T_SOURCE" IS 'Source system the record because of which this records was created, came from'
   COMMENT ON COLUMN "IL"."CASE"."T_PRIMARY_SYSTEM" IS 'System that initially created the record because of which this records was created'
   COMMENT ON COLUMN "IL"."CASE"."T_JOB_ID" IS 'ID of the job (smallest runnable unit) that touched this record.'
   COMMENT ON COLUMN "IL"."CASE"."T_BATCH_ID" IS 'ID of the batch (largest load unit) that touched this record.'
   COMMENT ON COLUMN "IL"."CASE"."FK_PTNT_T_RECORD_ID" IS 'Reference to the patient'
   COMMENT ON COLUMN "IL"."CASE"."FK_MPST_T_RECORD_ID" IS 'Reference to the case type mapping'
   COMMENT ON COLUMN "IL"."CASE"."CASE_ID" IS 'Identifcator of the source system'
   COMMENT ON COLUMN "IL"."CASE"."ADMISSION_DATE" IS 'Point in time when the first contact was recorded.'
   COMMENT ON COLUMN "IL"."CASE"."DISCHARGE_DATE" IS 'Point in time when the patient officially was discharged.'
   COMMENT ON COLUMN "IL"."CASE"."VALID_FROM" IS 'Point in time when this case record version has become valid in the source system'
   COMMENT ON COLUMN "IL"."CASE"."VALID_UNTIL" IS 'Point in time when this case record version has become invalid in the source system'
   COMMENT ON TABLE "IL"."CASE"  IS 'Stores occasions when a patient has seen a health care person be it stationary or ambulatory.
$Header: http://ictgitwp01.ms.uhbs.ch:18080/svn/cdwh/trunk/database/IL/dataloss_installable/tables/CASE.sql 83 2018-03-07 16:09:32Z kellnert $'
```

**Comment by GiantCrocodile** [comment]
@thiemo-taken You have some kind of url in your example data. Remove it, if it's sensitive.

**Comment by serge-rider** [comment]
Perhaps this is somehow related to Oracle driver version.
I suspect that something went wrong during connection initialization.
Are there any other error in error/debug logs?

**Comment by thiemo-taken** [comment]
Actually, I have some problems with jdbc connections erratically not being able to connect (sqlplus on the same computer works flawlessly. I use TNS. Please find attached the log file of a fresh started with log wiped beforehand.
[dbeaver_file.log](https://github.com/dbeaver/dbeaver/files/1806297/dbeaver_file.log)

**Comment by serge-rider** [comment]
I've added a couple of fixes, hopefully they will resolve this issue.

BTW I see many "connection refused" errors in logs. Looks like failed attempt to reestablish connection. This may be the actual cause of this error (NullPointerException). Does this error happen always or only after connection failures?

**Comment by thiemo-taken** [comment]
As mentioned before I do have problems with connection and I placed the issue with our dbas. They did not get back to me with respect to the issue. However, I would not expect to have trouble because of the just for opening a dialog.
I cannot compile here at work, so I shall wait for a binary release to test.

**Comment by thiemo-taken** [comment]
I used to use the thin driver of ojdbc8.jar. I switched to the OCI driver of ojdbc6.jar and I have no problems anylonger. Thin driver of later package seems to not have those connection problems though I did only connect onece.
